### PR TITLE
Implement fmt::Debug on all types

### PR DESCRIPTION
**Pull request by kamalmarhubi**
This vastly increases ease of experimentation with the libraries.

### CODE REVIEW

File: rust-install/src/component/package.rs [hunk 1]
```diff
@@ -31,6 +32,10 @@ pub trait Package {
                    -> Result<Transaction<'a>>;
 }
 
+pub trait PackageDebug: Package + fmt::Debug {}
+impl<T: Package + fmt::Debug> PackageDebug for T {}
```
Comment (Diggsey) [inline rust-install/src/component/package.rs]: Instead of introducing `PackageDebug`, can we just have `Package` imply `Debug`.

**Comment by Diggsey** [comment]
LGTM with that one change!

**Comment by kamalmarhubi** [comment]
Do you want me to squash?

File: rust-install/src/component/package.rs (hunk 1 above)
Comment (kamalmarhubi) [inline rust-install/src/component/package.rs]: Done.

**Comment by Diggsey** [comment]
No, it's fine thanks!

**Pull request merged**